MechJeb2/PartExtensions.cs

```diff
@@ -102,31 +102,17 @@ public static bool IsSepratron(this Part p)
 
         public static bool IsEngine(this Part p)
         {
-            for (int i = 0; i < p.Modules.Count; i++)
-            {
-                PartModule m = p.Modules[i];
-                if (m is ModuleEngines) return true;
-            }
-            return false;
+            return p.FindModuleImplementing<ModuleEngines>() != null;

```

Comment:
Ah, good point @sarbian -- will do. Btw, what are good ways to test memory allocations in KSP/Unity?